pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -1017,3 +1026,26 @@ func (r *Reconciler) pvsForProvisioner(ctx context.Context, cluster *kubermaticv
 	}
 	return pvsForProvisioner, nil
 }
+
+func (r *Reconciler) cleanupDefaultStorageClassAddon(ctx context.Context, cluster *kubermaticv1.Cluster, addon *kubermaticv1.Addon) error {
+	cl, err := r.KubeconfigProvider.GetClient(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to get client for usercluster: %w", err)
+	}
+	addonResourcesListOption := &ctrlruntimeclient.ListOptions{
+		Raw: &metav1.ListOptions{
+			LabelSelector: addonLabelKey + "=" + addon.Spec.Name,
+		},
+	}
+	// delete storageclass created by "default-storage-addon"
+	scList := &storagev1.StorageClassList{}
+	if err := cl.List(ctx, scList, addonResourcesListOption); err != nil {
+		return fmt.Errorf("failed to list storageclass : %w", err)
+	}
+	for i := 0; i < len(scList.Items); i++ {
+		if err := cl.Delete(ctx, &scList.Items[i]); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete storageclass: %w", err)
+		}
+	}
```

Comment:
Can't this be a single .Delete() call with a label selector?

Reply:
Delete options doesn't have a label selector - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.2/pkg/client#DeleteOptions

Reply:
```go
sc := &storagev1.StorageClass{}
client.DeleteAllOf(ctx, sc, ctrlruntimeclient.MatchingLabels{
	"addonLabelKey": "foobar",
})
```

This could work, or maybe the heat today is too much for me? :sweat_smile:

Reply:
Wasn't aware of this method, works fine.